In [1]:
import numpy as np 
import pandas as pd

<div style="font-size: 18px; font-weight: bold;">Task 1. Data Quality</div>

a. Provide an example of poor quality structured data

b. How would you recognize this poor quality? Write 3 - 4 sentences that show how the data fails to include properties of good quality data.

c. Provide an example of poor quality unstructured data

d. Unstructured data can be more difficult to assess than structured data. Just as you did in part b, write 3 - 4 sentences that show how this unstructured data fails to check requirements of good quality data.

<div style="font-size: 18px; font-weight: bold;">a. Example of poor quality structured data</div>

In [2]:
#Source:  https://www.kaggle.com/datasets/suriyaganesh/resume-dataset-structured?select=01_people.csv

# Load dataset  
file_path = r"C:\Users\kaleu\Desktop\WQU\GWP_Q1\01_people.csv"
people = pd.read_csv(file_path)  

# Display dataset preview  
people.head()

person_id                                               name email phone  \
0          1                             Database Administrator   NaN   NaN   
1          2                             Database Administrator   NaN   NaN   
2          3                      Oracle Database Administrator   NaN   NaN   
3          4  Amazon Redshift Administrator and ETL Develope...   NaN   NaN   
4          5             Scrum Master Scrum Master Scrum Master   NaN   NaN   

  linkedin  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN

In [3]:
missing = people.isnull().sum()
missing_percent = (missing / len(people) * 100).round(2)
missing_percent

person_id     0.00
name          0.21
email        97.10
phone        96.66
linkedin     84.46
dtype: float64

In [4]:
print(people.info())
print(people.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54933 entries, 0 to 54932
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   person_id  54933 non-null  int64 
 1   name       54819 non-null  object
 2   email      1593 non-null   object
 3   phone      1833 non-null   object
 4   linkedin   8538 non-null   object
dtypes: int64(1), object(4)
memory usage: 2.1+ MB
None
(54933, 5)


<div style="font-size: 18px; font-weight: bold;">Table Schemas</div>

| Column Name | Data Type     | Description                        | Constraints             | Example                     |
|-------------|--------------|------------------------------------|-------------------------|-----------------------------|
| person_id   | INTEGER      | Unique identifier for each person  | Primary Key, Not Null   | 1                           |
| name        | VARCHAR(255) | Full name of the person            | May be Null             | "Database Administrator"    |
| email       | VARCHAR(255) | Email address                      | May be Null             | "john.doe@email.com"        |
| phone       | VARCHAR(50)  | Contact number                     | May be Null             | "+1-555-0123"               |
| linkedin    | VARCHAR(255) | LinkedIn profile URL               | May be Null             | "linkedin.com/in/johndoe"   |


<div style="font-size: 18px; font-weight: bold;">b. How would you recognize this poor quality? </div>

When we first examined the dataset, we observed that many of the key fields were blank: columns such as email(97%missing), phone(97%missing), and LinkedIn(85%missing), which immediately made the table feel **incomplete**. As we looked closer, we noticed the name column is **inconsistent**, often containing job titles instead of individuals' names. The name column also contain **inaccurate** entry like one row listed “Scrum Master” three times. These are exactly the kinds of problems that Domo (2023) and Freeman (2024) describe when discussing poor data quality: missing values, inconsistent records, and information that can’t really be trusted.